### ISSUE TITLE
Empty xmlns="" attribute when encoding with etree_element_class=lxml.etree.Element

### ISSUE THREAD

**Issue by danifus** [sissaschool/xmlschema#252]
When encoding data using `etree_element_class=lxml.etree.Element`, the generated xml has an empty  `xmlns=""` attribute.

```
        elem = self.col_schema.encode(
            xd,
            path="./col:collection",
            namespaces=self.col_namespaces,
            etree_element_class=lxml.etree.Element,
        )
```

Note the `xmlns=""` attribute in the following:
```
<col:collection xmlns:col="http://example.com/ns/collection" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="" xsi:schemaLocation="http://example.com/ns/collection collection.xsd">
    ...
</col:collection>
```

I've bisected the commits to find that this was introduced in https://github.com/sissaschool/xmlschema/commit/4593e4cd42712970f4dfda86bd4ec39af208b2d8

Specifically the changes in this method: https://github.com/sissaschool/xmlschema/commit/4593e4cd42712970f4dfda86bd4ec39af208b2d8#diff-cc48914b2c2b191c2a8d65a358a59505a57a65051ef627e8280f10fdb7949635R128

The `"": ""` entry is added here: https://github.com/sissaschool/xmlschema/blob/master/xmlschema/xpath.py#L193

It can be fixed by changing this line:
`self._namespaces = namespaces`
to:
`self._namespaces = namespaces.copy()`

but I'm not sure what the appropriate fix is here.

This `"": ""` entry in `namespaces` is present for all element classes but the empty `xmlns=""` attribute only appears when using the lxml element class because the namespaces from `self.namespaces` are only attached to the root element when using the lxml element class (https://github.com/sissaschool/xmlschema/blob/master/xmlschema/converters/default.py#L228-L230). Python's `xml.etree` elements handle namespaces differently and are unaffected by this.

**Comment by brunato** [comment]
Hi @danifus,
i'm going to merge your PR with the additional failing test and check this in details.

At first sight i think that https://github.com/sissaschool/xmlschema/blob/master/xmlschema/xpath.py#L193 is correct but has the side effect of generate this issue. The resolution maybe is in skipping adding empty uri if also the prefix is empty, with an extra check.

Thank you

**Comment by brunato** [comment]
... well, I will fix and then merge the PR. 

In any case `xmlns=""` is redundant but legal (instead `xmlns:pfx=""` is illegal)

**Comment by danifus** [comment]
Thanks for looking at it. You're right that it is legal but I was just being a perfectionist :p

Thanks for all your work on this project! It's a great help!

**Comment by brunato** [comment]
Many thank also to you! Your proposals and suggestions about XML resource access were fundamental to improve the security of this library!

**Comment by brunato** [comment]
Issue fixed and PR #253 merged. I keep this issue open until the next package release is completed and published.

Thank you

**Comment by brunato** [comment]
Hi @danifus,
a new release is available.

Bye
